pytorch_lightning/profiler/pytorch.py

```diff
@@ -448,68 +613,242 @@ def on_trace_ready(profiler):
-        if self._emit_nvtx:
-            self._parent_profiler = self._create_profiler(torch.cuda.profiler.profile)
-            self.profiler = self._create_profiler(torch.autograd.profiler.emit_nvtx)
+class PyTorchProfilerKineto(BasePyTorchProfiler):
+    """Use new lower level PyTorch profiler API torch.profiler._KinetoProfile to simplify the profiler
+    implementation by removing the schedule wrapper."""
+
+    def __init__(
+        self,
+        dirpath: Optional[Union[str, Path]] = None,
+        filename: Optional[str] = None,
+        group_by_input_shapes: bool = False,
+        emit_nvtx: bool = False,
+        export_to_chrome: bool = True,
+        row_limit: int = 20,
+        sort_by_key: Optional[str] = None,
+        record_module_names: bool = True,
+        wait_step: int = 1,
+        warmup_step: int = 1,
+        active_step: int = 3,
+        **profiler_kwargs: Any,
+    ) -> None:
+        """This profiler uses PyTorch's Autograd Profiler and lets you inspect the cost of.
+
+        different operators inside your model - both on the CPU and GPU
+
+        Args:
+            dirpath: Directory path for the ``filename``. If ``dirpath`` is ``None`` but ``filename`` is present, the
+                ``trainer.log_dir`` (from :class:`~pytorch_lightning.loggers.tensorboard.TensorBoardLogger`)
+                will be used.
+
+            filename: If present, filename where the profiler results will be saved instead of printing to stdout.
+                The ``.txt`` extension will be used automatically.
+
+            group_by_input_shapes: Include operator input shapes and group calls by shape.
+
+            emit_nvtx: Context manager that makes every autograd operation emit an NVTX range
+                Run::
+
+                    nvprof --profile-from-start off -o trace_name.prof -- <regular command here>
+
+                To visualize, you can either use::
+
+                    nvvp trace_name.prof
+                    torch.autograd.profiler.load_nvprof(path)
+
+            export_to_chrome: Whether to export the sequence of profiled operators for Chrome.
+                It will generate a ``.json`` file which can be read by Chrome.
+
+            row_limit: Limit the number of rows in a table, ``-1`` is a special value that
+                removes the limit completely.
+
+            sort_by_key: Attribute used to sort entries. By default
+                they are printed in the same order as they were registered.
+                Valid keys include: ``cpu_time``, ``cuda_time``, ``cpu_time_total``,
+                ``cuda_time_total``, ``cpu_memory_usage``, ``cuda_memory_usage``,
+                ``self_cpu_memory_usage``, ``self_cuda_memory_usage``, ``count``.
+
+            record_module_names: Whether to add module names while recording autograd operation.
+
+            profiler_kwargs: Keyword arguments for the PyTorch profiler. This depends on your PyTorch version
```

Comment:
Is this still True? This class is now only used for a specific PyTorch version

Reply:
yes, the new profiler class's behavior is same with the legacy profiler. And _record_module_names is used in start method.